File: python/2401. Longest Nice Subarray.py
from typing import List

class Solution:
    def longestNiceSubarray(self, nums: List[int]) -> int:
        acc = 0

        result = 0
        left = 0
        n = len(nums)

        for right in range(n):
            num = nums[right]
            while num & acc:
                acc = acc ^ nums[left]
                left += 1
            acc = acc | num
            result = max(result, right - left + 1)
        return result

File: rust/2401. Longest Nice Subarray.rs
impl Solution {
    pub fn longest_nice_subarray(nums: Vec<i32>) -> i32 {
        let mut acc = 0u32;
        let mut l = 0usize;
        let mut result = 0i32;
        let n = nums.len();

        for r in 0..n {
            let num = nums[r] as u32;
            while (acc & num) != 0 {
                acc ^= nums[l] as u32;
                l += 1;
            }
            acc |= num;
            result = result.max((r - l + 1) as i32);
        }
        result
    }
}
